webhooks-extension/endpoints/credentials.go

```diff
@@ -0,0 +1,261 @@
+	creds := []credential{}
+	emptyCred := credential{}
+	for _, secret := range secrets.Items {
+		cred := secretToCredential(&secret, true)
+		if cred != emptyCred {
+			creds = append(creds, cred)
+			logging.Log.Infof("getAllCredentials Found credential %+v\n", cred)
+		}
+	}
+
+	logging.Log.Infof("getAllCredentials returning +%v", creds)
+
+	// Write the response
+	response.AddHeader("Content-Type", "application/json")
+	response.WriteEntity(creds)
+}
+
+// Convert credential struct into K8s secret struct
+func credentialToSecret(cred credential, namespace string, response *restful.Response) *corev1.Secret {
+	// Create new secret struct
+	secret := corev1.Secret{}
+	secret.Type = corev1.SecretTypeOpaque
+	secret.SetNamespace(namespace)
+	secret.SetName(cred.Name)
+	secret.Data = make(map[string][]byte)
+	secret.Data["accessToken"] = []byte(cred.AccessToken)
+	if cred.SecretToken != "" {
+		secret.Data["secretToken"] = []byte(cred.SecretToken)
+	} else {
+		secret.Data["secretToken"] = getRandomSecretToken()
+	}
+	return &secret
+}
+
+var (
+	src = rand.NewSource(time.Now().UnixNano())
+)
+
+const (
+	tokenBytes   = "123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	tokenIdxBits = 6                   // 6 bits = 2^6 = 64 characters in tokenBytes
+	tokenIdxMask = 1<<tokenIdxBits - 1 // All 1-bits, as many as tokenIdxBits
+)
+
+// Generate a random 20-character string, returned as []byte.
+// With thanks to https://medium.com/@kpbird/golang-generate-fixed-size-random-string-dd6dbd5e63c0
+func getRandomSecretToken() []byte {
+	b := make([]byte, 20)
+	for i := 0; i < 20; {
+		idx := int(src.Int63() & tokenIdxMask)
+		if idx < len(tokenBytes) {
+			b[i] = tokenBytes[idx]
+			i++
+		}
+	}
+	return b
+}
+
+// Convert K8s secret struct into credential struct
+func secretToCredential(secret *corev1.Secret, mask bool) credential {
+	var cred credential
+	if secret.Data["accessToken"] != nil {
+		cred = credential{
+			Name:            secret.GetName(),
+			AccessToken:     string(secret.Data["accessToken"]),
+			SecretToken:     string(secret.Data["secretToken"]),
+			ResourceVersion: secret.GetResourceVersion(),
+		}
+		if mask {
+			cred.AccessToken = "********"
+		}
+	}
+	return cred
+}
+
+// Checks the Accept header and reads the content into the entityPointer.
+func getQueryEntity(entityPointer interface{}, request *restful.Request, response *restful.Response) (err error) {
+	if err := request.ReadEntity(entityPointer); err != nil {
+		errorMessage := "Error parsing request body."
+		utils.RespondMessageAndLogError(response, err, errorMessage, http.StatusBadRequest)
+		return err
+	}
+	return nil
+}
+
+func (r Resource) verifyCredentialParameters(cred credential, response *restful.Response) bool {
+	errorMessage := ""
+	if cred.Name == "" {
+		errorMessage = fmt.Sprintf("Error: Name must be specified")
+	} else if cred.AccessToken == "" {
+		errorMessage = fmt.Sprintf("Error: AccessToken must be supplied")
```

Comment:
inconsistency in this message and that on 225 .... I'd use specified in both cases.